index.html

```diff
@@ -101,19 +199,623 @@ <h3>
+          is locked and the socket cannot be closed. This forces the developer to
+          decide how to handle any read or write operations that are in progress.
+
+
+          For example, to ensure that all buffered data has been transmitted before
+          the socket is closed the application must await the {{Promise}} returned by
+          `writer.`{{WritableStreamDefaultWriter/close()}}.
+
+          <pre class="js">
+            const encoder = new TextEncoder();
+            const { writable } = await socket.connection;
+            const writer = writable.getWriter();
+
+            writer.write(encoder.encode("A long message that will take..."));
+            await writer.close();
+            await socket.close();
+          </pre>
+
+          To discard any unsent data the application could instead call
+          `writer.`{{WritableStreamDefaultWriter/abort()}}.
+
+          If a loop is being used to read data from the socket, then it must be exited before calling `socket`.{{TCPSocket/close()}}.
+
+          <pre class="js">
+            const { readable } = await socket.connection;
+            const reader = readable.getReader();
+
+            async function readUntilClosed() {
+              while (true) {
+                const { value, done } = await reader.read();
+                if (done) {
+                  // |reader| has been canceled.
+                  break;
+                }
+                // Do something with |value|...
+              }
+
+              await socket.close();
+            }
+
+            const read_complete = readUntilClosed();
+
+            // Sometime later...
+            reader.cancel();
+            await read_complete;
+          </pre>
+
+        </aside>
+
+        The {{TCPSocket/close()}} method steps are:
+
+        <ol>
+          <li>
+            If [=this=].{{TCPSocket/[[connectionPromise]]}} is rejected or not yet resolved, [=reject=] with
+            "{{InvalidStateError}}" {{DOMException}}.
+          <li>
+            If [=this=].{{TCPSocket/[[closedPromise]]}} is already resolved, return [=this=].{{TCPSocket/[[closedPromise]]}}.
+          <li>
+            If [=this=].{{TCPSocket/[[readable]]}} or [=this=].{{TCPSocket/[[writable]]}} are <a href="https://streams.spec.whatwg.org/#lock">locked</a>,
+            [=reject=] with "{{InvalidStateError}}" {{DOMException}}.
+          <li>
+            Let |cancelPromise:Promise| be the result of invoking
+            [=ReadableStream/cancel=] on [=this=].{{TCPSocket/[[readable]]}}.
+          <li>
+            Mark |cancelPromise| as handled.
+          <li>
+            Let |abortPromise:Promise| be the result of invoking
+            [=WritableStream/abort=] on [=this=].{{TCPSocket/[[writable]]}}.
+          <li>
+            Mark |abortPromise| as handled.
+          <li>
+            [=Resolve=] [=this=].{{TCPSocket/[[closedPromise]]}} with `undefined`.
```

Comment:
Won't this be resolved by the steps which handle the streams being closed?

Reply:
True!